Mathlib/SetTheory/Game/Impartial.lean

```diff
@@ -68,49 +69,47 @@ instance moveRight_impartial {G : PGame} [h : G.Impartial] (j : G.RightMoves) :
     (G.moveRight j).Impartial :=
   (impartial_def.1 h).2.2 j
 
-theorem impartial_congr : ∀ {G H : PGame} (_ : G ≡r H) [G.Impartial], H.Impartial
-  | G, H => fun e => by
-    intro h
-    exact impartial_def.2
-      ⟨Equiv.trans e.symm.equiv (Equiv.trans (neg_equiv_self G) (neg_equiv_neg_iff.2 e.equiv)),
-        fun i => impartial_congr (e.moveLeftSymm i), fun j => impartial_congr (e.moveRightSymm j)⟩
-termination_by G H => (G, H)
-
-instance impartial_add : ∀ (G H : PGame) [G.Impartial] [H.Impartial], (G + H).Impartial
-  | G, H, _, _ => by
-    rw [impartial_def]
-    refine ⟨Equiv.trans (add_congr (neg_equiv_self G) (neg_equiv_self _))
-        (Equiv.symm (negAddRelabelling _ _).equiv), fun k => ?_, fun k => ?_⟩
-    · apply leftMoves_add_cases k
-      all_goals
-        intro i; simp only [add_moveLeft_inl, add_moveLeft_inr]
-        apply impartial_add
-    · apply rightMoves_add_cases k
-      all_goals
-        intro i; simp only [add_moveRight_inl, add_moveRight_inr]
-        apply impartial_add
-termination_by G H => (G, H)
-
-instance impartial_neg : ∀ (G : PGame) [G.Impartial], (-G).Impartial
-  | G, _ => by
-    rw [impartial_def]
-    refine ⟨?_, fun i => ?_, fun i => ?_⟩
-    · rw [neg_neg]
-      exact Equiv.symm (neg_equiv_self G)
-    · rw [moveLeft_neg']
-      apply impartial_neg
-    · rw [moveRight_neg']
-      apply impartial_neg
-termination_by G => G
+theorem impartial_congr {G H : PGame} (e : G ≡r H) [G.Impartial] : H.Impartial :=
+  impartial_def.2
+    ⟨Equiv.trans e.symm.equiv (Equiv.trans (neg_equiv_self G) (neg_equiv_neg_iff.2 e.equiv)),
+      fun i => impartial_congr (e.moveLeftSymm i), fun j => impartial_congr (e.moveRightSymm j)⟩
+termination_by G
+
+instance impartial_add (G H : PGame) [G.Impartial] [H.Impartial] : (G + H).Impartial := by
+  rw [impartial_def]
+  refine ⟨Equiv.trans (add_congr (neg_equiv_self G) (neg_equiv_self _))
```

Comment:
Looks like this line can be golfed with some dot notation, if you like

Reply:
Dot notation is broken on `PGame.Equiv` for whatever reason.

Reply:
Hmm, sounds like that should be fixed, or reported as an issue

Reply:
Looked into it, the issue is that the `≈` notation is setoid notation, rather than notation for `PGame.Equiv` itself.